qemu/tests/block_hotplug.py

```diff
@@ -177,32 +179,44 @@ def check_disk_size(did, excepted_size):
     vm.verify_alive()
 
     for iteration in range(int(params.get("repeat_times", 3))):
-        for index, img in enumerate(data_imgs):
-            data_devs = create_block_devices(img)
-            if need_plug:
-                new_disk = plug_block_devices('hotplug', data_devs).pop()
-
-                if windows:
-                    if iteration == 0:
-                        format_disk_win()
-                if need_check_disk_size:
-                    check_disk_size(new_disk if windows else new_disk[5:],
-                                    params['image_size_%s' % img])
-
-                if disk_op_cmd:
-                    run_io_test()
-            unplug_devs.extend(get_block_devices(
-                data_devs) if need_plug else get_block_devices(img))
-        if sub_test_after_plug:
-            run_sub_test(sub_test_after_plug)
-        if shutdown_after_plug:
-            return
-
-        if sub_test_before_unplug:
-            run_sub_test(sub_test_before_unplug)
-
-        plug_block_devices('unplug', unplug_devs)
-        del unplug_devs[:]
-
-        if sub_test_after_unplug:
-            run_sub_test(sub_test_after_unplug)
+        try:
+            for index, img in enumerate(data_imgs):
+                data_devs = create_block_devices(img)
+                if need_plug:
+                    new_disk = plug_block_devices('hotplug', data_devs).pop()
+
+                    if windows:
+                        if iteration == 0:
+                            format_disk_win()
+                    if need_check_disk_size:
+                        check_disk_size(new_disk if windows else new_disk[5:],
+                                        params['image_size_%s' % img])
+
+                    if disk_op_cmd:
+                        run_io_test()
+                unplug_devs.extend(get_block_devices(
+                    data_devs) if need_plug else get_block_devices(img))
+
+            if sub_test_after_plug:
+                run_sub_test(sub_test_after_plug)
+            if shutdown_after_plug:
+                return
+
+            if sub_test_before_unplug:
+                run_sub_test(sub_test_before_unplug)
+
+            plug_block_devices('unplug', unplug_devs)
+            del unplug_devs[:]
+
+            if sub_test_after_unplug:
+                run_sub_test(sub_test_after_unplug)
+        except Exception as e:
+            pid = vm.get_pid()
+            test.log.debug("Find %s Exception:'%s'.", pid, e)
+            if pid:
+                logdir = test.logdir
+                process.getoutput("gstack %s > %s/gstack.log" % (pid, logdir))
+                process.getoutput("timeout 20 strace -tt -T -v -f -s 32 -p %s -o %s/strace.log" % (pid, logdir))
```

Comment:
Recommend a maximum of 79 characters per line, suggest to update it as block_resize.py - line 256. Thanks.